lib/srv/server/ec2_watcher.go

```diff
@@ -131,9 +137,14 @@ func WithPollInterval(interval time.Duration) Option {
 // MakeEvents generates ResourceCreateEvents for these instances.
 func (instances *EC2Instances) MakeEvents() map[string]*usageeventsv1.ResourceCreateEvent {
 	resourceType := types.DiscoveredResourceNode
-	if instances.DocumentName == types.AWSAgentlessInstallerDocument {
+
+	switch {
+	case instances.DocumentName == types.AWSAgentlessInstallerDocument:
 		resourceType = types.DiscoveredResourceAgentlessNode
+	case instances.Integration != "":
+		resourceType = types.DiscoveredResourceEICENode
```

Comment:
This may be fine for now but I've a question about future-proofness of this. If/when we introduce support for similar GCP/Azure integrations, how do we determine what resource type this is? Cause right now we just assume that if it's any integration then it must be an EIC node.

Reply:
For GCP/Azure we should be good, because this method is only for EC2 instances.

However, we might have an issue if/when we develop another AWS integration subkind.
I don't think we should worry too much about it, but I think we'll another Node subkind if that integration connects to nodes using a different method.